src/components/Root.js

```diff
@@ -9,6 +9,9 @@ class Root extends DesktopComponent {
   constructor() {
     super();
     libui.startLoop();
+    libui.Ui.onShouldQuit(() => {
+      libui.stopLoop();
```

Comment:
Is more cleanup needed?

Reply:
> Is more cleanup needed?

I don't know how the proton internal works, but I think this is run for every components created?

Reply:
Does `UiWindow::onClosing` get called when `libui.stopLoop();` is run?

Reply:
> Does UiWindow::onClosing get called when libui.stopLoop(); is run?

`libui-node` does not call it. But maybe libui call it directly. @andlabs ?

Reply:
So it seems that `onClosing` only gets called when the window is closed and that `onShouldQuit` only gets called when the quit menu item is used. I could add `onShouldQuit` to App, and expose that to the user.

Reply:
Yes, because you can have multiple windows. Maybe this model is a bit confusing, but other models would be too restrictive. Not sure what the right balance is.